react/QuantitySelector.js

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+/**
+* The quantity selector component.
+*/
+class QuantitySelector extends Component {
+  constructor(props) {
+    super(props)
+  }
+
+  handleChange = (event) => {
+    const maxQuantity = this.props.maxQuantity
+    let quantity = event.target.value
+    if (quantity > maxQuantity) {
+      quantity = maxQuantity
+      this.props.onMaxReached()
+    } else if (quantity < 0) {
+      quantity = 0
+    }
+    this.props.updateTotalPrice(quantity)
+  }
+
+  render() {
+    const {maxQuantity, currentQuantity} = this.props
+    return (<input className="f4 dark-gray br2-l" type="number" name="quantity"
+      min="0" max={maxQuantity} value={currentQuantity} onChange={this.handleChange} />)
+  }
+}
+
+QuantitySelector.propTypes = {
+  /** The product's maximum quantity that the client can buy */
```

Comment:
Product's maximum ... ⛏

Reply:
Write just: 
"Current quantity to be setted as the initial value"
"Called when the client set the quantity selector"
"Define if can buy more items than the maximum limit"

Reply:
Done!